lib/Clam/CfgBuilder.cc

```diff
@@ -1961,6 +2000,36 @@ void CrabIntraBlockBuilder::doMemIntrinsic(MemIntrinsic &I) {
   m_bb.havoc(m_lfac.mkIntVar(32), valueToStr(I));
 }
 
+void CrabIntraBlockBuilder::doGepFatAssign(Instruction &I, unsigned width,
+                                           lin_exp_t offset) {
+  if (!isa<BitCastInst>(I) && !isa<GetElementPtrInst>(I)) {
+    return;
+  }
+  CRAB_LOG("cfg-fat", llvm::errs()
+                          << "Gep: add two extra variable for " << I << "\n");
+  Value *vlhs = llvm::cast<llvm::Value>(&I);
+  Value *vptr = I.getOperand(0);
+  auto it = m_ref_fat_map.find(vptr);
+  if (it == m_ref_fat_map.end()) {
+    CLAM_WARNING("Could not find gep ptr from map " << *vptr << "\n at" << I);
+  } else {
+    var_t ptr_obj_sz = it->second.first;
+    var_t ptr_offset = it->second.second;
+    var_t ref_obj_sz = m_lfac.mkIntVar(width);
+    var_t ref_offset = m_lfac.mkIntVar(64);
+    m_bb.assign(ref_obj_sz, ptr_obj_sz);
+    var_t tmp_offset = m_lfac.mkIntVar(width);
+    m_bb.assign(tmp_offset, offset);
+    var_t ext_offset = m_lfac.mkIntVar(64);
+    m_bb.zext(tmp_offset, ext_offset);
```

Comment:
We should be able to avoid `zero-extension` instruction. All the ghost variables used for offset and size should have the same bitwidth. I can see that you take as input `offset` which is already a crab linear expression (i.e., it has been already translated). It should have also the same bitwidth than offset and size. If not, we should fix that.